### PR TITLE
"Hello world" project with Rails, React and Webpack

### PR DESCRIPTION
**Pull request by DaveZag**
# "Hello world" project with Rails, React and Webpack
### Hi there ✋🏽 , on this branch I:
- Create a new Rails app called 'hello-rails-react'.
- Install needed node dependencies as described in the previous lesson's tutorial.
- Create a babel.config.js in the root folder of your project as described in the previous lesson's tutorial.
- Update webpack.config.js to include the babel loader as described in the previous lesson's tutorial.
- Make sure that your project has a Postgres database set up.
- Create a table for storing your messages and create 5 different greetings.
- Create an API endpoint that selects a random greeting from your table (you will need a controller with an action and Rails route).
- Create a static view that will be the root of your app.
- Create the App component with react-router and render it in your static view.
- Create the Greeting component that will display a greeting. Set it up as a route in your App component.
- Create a store, an action, and a reducer that will connect to your API endpoint to get the random greeting.
- Display the random greeting in your Greeting component.
### Preview:
image

### CODE REVIEW

**Comment by julie-ify** [comment]
Hi @DaveZag,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

### Kindly fix your linter errors

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by DaveZag** [comment]
Hi @julie-ify , Thanks for the feedback🙂

File: app/javascript/application.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import { Provider } from "react-redux";
+
+import Greeting from "./components/Greeting.js";
+
+function App() {
+  return (
+    <div>
+      <Greeting />
+    </div>
+  );
```
Comment (mikethreels) [inline app/javascript/application.js]: - [x] Routes should be setup using react-router-dom and the App component should have a route leading to the greeting component.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Dave, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- linters are passing

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/javascript/application.js (hunk 1 above)
Comment (DaveZag) [reply app/javascript/application.js]: Hi @mikethreels, thanks for the feedback. All required changes have been implemented.

File: app/javascript/application.js [hunk 2]
```diff
@@ -22,3 +25,13 @@ root.render(
...
+
+/*  Better version of this app to be uncommented after code review. NB: It won't be deleted.
+function App() {
+  return (
+    <div>
+      <Greeting />
+    </div>
+  );
+}
+*/
```
Comment (Lucas-Masaba) [inline app/javascript/application.js:37]: [OPTIONAL]
- Please ensure not leave unnecessary comments in your code.

File: app/javascript/components/Greeting.js [hunk 3]
```diff
@@ -10,16 +15,16 @@ const Greeting = () => {
     dispatch(fetchGreetings());
   }, []);
 
-  const newGreeting = () => {
-    dispatch(fetchGreetings());
-  };
+  // const newGreeting = () => {
+  //   dispatch(fetchGreetings());
+  // };
```
Comment (Lucas-Masaba) [inline app/javascript/components/Greeting.js:20]: [OPTIONAL]
- Please ensure not leave unnecessary comments in your code.

File: app/javascript/components/Greeting.js [hunk 4]
```diff
@@ -10,16 +15,16 @@ const Greeting = () => {
...
 
   return (
     <div>
       <h1 className="message">{greeting.text}</h1>
-      <button onClick={newGreeting} type="button">
+      {/* <button onClick={newGreeting} type="button">
         Generate
-      </button>
+      </button> */}
```
Comment (Lucas-Masaba) [inline app/javascript/components/Greeting.js:27]: [OPTIONAL]
- Please ensure not leave unnecessary comments in your code.

**Review by Lucas-Masaba — APPROVED** [review]
# Hello @DaveZag  👋 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

almost-there

## Highlights 🍾 
- Linter checks passing ✔️ 
- Thanks for making changes as requested by the previous reviewer  👍 

## Optional suggestions 💭 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏 💻 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Lucas-Masaba in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**